ext/vulnsrc/redhat/redhat.go

```diff
@@ -31,40 +32,47 @@ import (
 	"github.com/quay/clair/v3/database"
 	"github.com/quay/clair/v3/ext/versionfmt/modulerpm"
 	"github.com/quay/clair/v3/ext/versionfmt/rpm"
+	"github.com/quay/clair/v3/ext/vulnsrc"
 	"github.com/quay/clair/v3/pkg/commonerr"
+	"github.com/quay/clair/v3/pkg/envutil"
 	"github.com/quay/clair/v3/pkg/httputil"
-	"github.com/quay/clair/v3/ext/vulnsrc"
 	log "github.com/sirupsen/logrus"
 )
 
 const (
-	// PulpV2BaseURL - base url for pulp v2 content
-	PulpV2BaseURL            = "https://www.redhat.com/security/data/oval/v2/"
 	// PulpManifest - url suffix for pulp manifest file
-	PulpManifest             = "PULP_MANIFEST"
+	PulpManifest = "PULP_MANIFEST"
 	// DbManifestEntryKeyPrefix - key prefix used to create flag for manifest entry hash key/value
 	DbManifestEntryKeyPrefix = "oval.v2.pulp.manifest.entry."
 	// DbLastAdvisoryDateKey - key prefix used to create flag for last advisory date key/value
-	DbLastAdvisoryDateKey    = "oval.v2.advisory.date.issued"
+	DbLastAdvisoryDateKey = "oval.v2.advisory.date.issued"
 	// DefaultLastAdvisoryDate - literal date (in case no existing last advisory date is found)
-	DefaultLastAdvisoryDate  = "1970-01-01"
+	DefaultLastAdvisoryDate = "1970-01-01"
 	// AdvisoryDateFormat date format for advisory dates ('magical reference date' for datetime format)
-	AdvisoryDateFormat       = "2006-01-02"
+	AdvisoryDateFormat = "2006-01-02"
 	// UpdaterFlag - key used for flag for updater
-	UpdaterFlag              = "RedHatOvalV2Updater"
+	UpdaterFlag = "redHatUpdater"
 	// UpdaterFlagDateFormat - date format for updater flag dates ('magical reference date' for datetime format)
-	UpdaterFlagDateFormat    = "2006-01-02 15:04:05"
+	UpdaterFlagDateFormat = "2006-01-02 15:04:05"
 	// AffectedType - affected type
-	AffectedType             = database.BinaryPackage
+	AffectedType = database.BinaryPackage
 	// CveURL - url for cve content
-	CveURL                   = "https://access.redhat.com/security/cve/"
+	CveURL = "https://access.redhat.com/security/cve/"
 )
 
+// OvalV2BaseURL - base url for oval v2 content
+var OvalV2BaseURL = envutil.GetEnv("OVAL_V2_URL", "https://www.redhat.com/security/data/oval/v2/")
+
 // SupportedArches - supported architectures
-var SupportedArches = map[string]bool { "x86_64":true, "noarch":true }
+var SupportedArches = map[string]bool{"x86_64": true, "noarch": true}
 
 // SupportedDefinitionTypes - supported definition classes
-var SupportedDefinitionTypes = map[string]bool { "patch":true }
+var SupportedDefinitionTypes = map[string]bool{"patch": true}
+
+// pendingVulnNames - quick running reference (by name) of all the vulnerabilities which have been added to the pending list
+var pendingVulnNames = map[string]bool{}
+
+var accumulatedVulnerabilities = []database.VulnerabilityWithAffected{}
```

Comment:
Why do you keep this on a global level? Having a global variable that can be huge is a very bad practice and this use case can be solved by just using local variables and do the merge in `Update()` instead in `CollectVulnsForAdvisory()`

Reply:
@Allda that is a good point, let me review the vuln-feature merge strategy ([lines 280-290](https://github.com/johnbe11/clair/blob/CWF-129_fix-rebase-2_updateintegration_tests/ext/vulnsrc/redhat/redhat.go#L280)) and find a better way to handle that.